tests/e2e/training-page-test.js

```diff
@@ -0,0 +1,44 @@
+import createServer from '../createServer'
+import { Selector, ClientFunction } from 'testcafe'
+require('dotenv').config()
+
+const hostname = `localhost`
+const port = `3001`
+const baseUrl = `${hostname}:${port}`
+const trainingPageUrl = `${baseUrl}/training`
+const getWindowLocation = ClientFunction(() => window.location)
+
+let server
+let firstModalLink
+fixture`Training page`.page`${trainingPageUrl}`
+  .before(async t => {
+    server = createServer(port)
+  })
+  .beforeEach(async t => {
+    firstModalLink = await Selector('a[data-testid^="modal"]').nth(0)
+  })
+  .after(() => server.close())
+
+test('should open the correct training course modal when accessed via a link', async t => {
+  const coursePathName = await firstModalLink
+    .getAttribute('data-testid')
+    .then(data =>
```

Comment:
I think the `data-testid` is useful for targeting specific nodes but they shouldn't contain the logic we're trying to test.

What I mean by this is that using this id for building the course path name doesn't seem right to me.

I'm no QA expert, but the test I'd create for this would be:
- navigate directly to the course link
- save some data about it (URL, page title, etc)
- click 'close
- click on the link for that same course and compare those URL, page title etc to make sure they're the same

Reply:
I completely agree. I don't like this test relying on how we form our `data-testid`.

The problem that we have is
> navigate directly to the course link

this assumes that we have a course link that's never going to change. Whereas our links are dynamically generated.

What if Greta removes the link that we hard code into this test, from the course catalog in Contentful?

Reply:
then the tests breaks. Isn't that what we want?

Reply:
Only if the test we are writing is trying to ensure that "node fundamentals" remains on the page as part of our Course Catalog.

If we want to allow Greta to change the catalog to whatever she wants, then we need a test that doesn't care _what_ the course is, but just cares about whether a link on the training page takes the user to a modal relating to that course.

Does that make sense?

Reply:
yes!